tests/test_ast.py

```diff
@@ -6,19 +6,31 @@
 def test_lite_repr():
     one = claripy.BVV(1, 8)
     two = claripy.BVV(2, 8)
-    a = claripy.BVS('a', 8)
-    b = claripy.BVS('b', 8)
+    a = claripy.BVS('a', 8, explicit_name=True)
+    b = claripy.BVS('b', 8, explicit_name=True)
 
-    nose.tools.assert_regex((a + one * b + two).shallow_repr(),
-                            r'\<BV8 __add__\(a_\d+_8, 1 \* b_\d+_8, 2\)\>')
-    nose.tools.assert_regex(((a + one) * (b + two)).shallow_repr(),
-                            r'\<BV8 \(a_\d+_8 \+ 1\) \* \(b_\d+_8 \+ 2\)\>')
-    nose.tools.assert_regex((a * one + b * two).shallow_repr(),
-                            r'\<BV8 a_\d+_8 \* 1 \+ b_\d+_8 \* 2\>')
-    nose.tools.assert_regex(((one + a)*(two + b)+(two + a)*(one + b)).shallow_repr(),
-                            r'\<BV8 \(1 \+ a_\d+_8\) \* \(2 \+ b_\d+_8\) \+ \(2 \+ a_\d+_8\) \* \(1 \+ b_\d+_8\)\>')
+    nose.tools.assert_equal((a + one * b + two).shallow_repr(),
+                            '<BV8 __add__(a, 1 * b, 2)>')
+    nose.tools.assert_equal(((a + one) * (b + two)).shallow_repr(),
+                            '<BV8 (a + 1) * (b + 2)>')
+    nose.tools.assert_equal((a * one + b * two).shallow_repr(),
+                            '<BV8 a * 1 + b * 2>')
+    nose.tools.assert_equal(((one + a)*(two + b)+(two + a)*(one + b)).shallow_repr(),
+                            '<BV8 (1 + a) * (2 + b) + (2 + a) * (1 + b)>')
 
 
+def test_associativity():
+    x = claripy.BVS('x', 8, explicit_name=True)
+    y = claripy.BVS('y', 8, explicit_name=True)
+    z = claripy.BVS('z', 8, explicit_name=True)
+    w = claripy.BVS('w', 8, explicit_name=True)
+
+    nose.tools.assert_equal((x - y - z - w).shallow_repr(),
+                            (x - (y - (z - w))).shallow_repr())
```

Comment:
this test is not correct! `x - y - z - w` is the same as `+x + -y + -z + -w`, while `x - (y - (z - w))` is the same as `+x + -y + +z + -w`

Reply:
Oops! You're right. I've missed that detail. I am gonna fix it.

Reply:
I've misunderstood your previous comment. Thanks for pointing the mistake. I've fixed the associativity tests and the code to handle it correctly.